### PR TITLE
chore(in-app-messaging): Add unit tests for carousel

### PR DESCRIPTION
**Pull request by cshfang**
#### Description of changes

This PR adds unit tests to the In-App Messaging Carousel Message component along with some minor edge case fixes discovered as a result of adding unit tests.

#### Description of how you validated changes
Tested code change in local sample app to make sure behavior was unchanged. 

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/aws-amplify-react-native/src/InAppMessaging/components/CarouselMessage/CarouselMessageItem.tsx [hunk 1]
```diff
@@ -18,9 +18,9 @@ import { FullScreenContent } from '../FullScreenMessage';
 import { useMessageProps } from '../hooks';
 
 import { getStyles } from './styles';
-import { CarouselMessageProps } from './types';
+import { CarouselMessageItemProps } from './types';
 
-export default function CarouselItem(props: CarouselMessageProps) {
+export default function CarouselMessageItem(props: CarouselMessageItemProps) {
```
Comment (calebpollman) [inline packages/aws-amplify-react-native/src/InAppMessaging/components/CarouselMessage/CarouselMessageItem.tsx:23]: 👍🏽

File: packages/aws-amplify-react-native/src/InAppMessaging/ui/Carousel/CarouselPageIndicator.tsx [hunk 2]
```diff
@@ -27,10 +27,10 @@ export default function CarouselPageIndicator({
 }: CarouselPageIndicatorProps) {
 	const items = useMemo(
 		() =>
-			new Array(numberOfItems)
+			new Array(numberOfItems ?? 0)
```
Comment (calebpollman) [inline packages/aws-amplify-react-native/src/InAppMessaging/ui/Carousel/CarouselPageIndicator.tsx:30]: Think what you have here is fine, but just for the sake of conversation would it make sense to set default values for `numberOfItems` and `currentIndex` of `0`, **or** just early return an empty array if `numberOfItems` is falsy?

**Review by Samaritan1011001 — APPROVED** [review]
Looks good, thanks!

File: packages/aws-amplify-react-native/src/InAppMessaging/components/CarouselMessage/CarouselMessage.tsx [hunk 3]
```diff
@@ -32,7 +33,9 @@ export default function CarouselMessage(props: CarouselMessageProps) {
 		};
 	}, [style]);
 
-	const renderItem = ({ item }) => <CarouselItem {...item} {...rest} />;
+	const renderItem = ({ item }: { item: InAppMessageComponentContentProps }) => (
+		<CarouselMessageItem {...item} {...rest} />
+	);
 
 	return (
 		<MessageWrapper disableSafeAreaView>
```
Comment (chintannp) [inline packages/aws-amplify-react-native/src/InAppMessaging/components/CarouselMessage/CarouselMessage.tsx:41]: is `disableSafeAreaView` intended for full screen view?

Comment (cshfang) [reply packages/aws-amplify-react-native/src/InAppMessaging/components/CarouselMessage/CarouselMessage.tsx:41]: The `disableSafeAreaView` prop in the MessageWrapper removes the SafeAreaView from that component since Carousel puts up a SAV per "slide".
